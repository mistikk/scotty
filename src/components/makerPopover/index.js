export { default as MarkerPopover } from './makerPopover.view';
