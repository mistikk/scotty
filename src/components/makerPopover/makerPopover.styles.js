import EStyleSheet from 'react-native-extended-stylesheet';

export default EStyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    paddingHorizontal: 20,
  },
  title: {
    fontSize: 20,
    marginBottom: 10,
  },
  button: {
    marginTop: 10,
    width: '100%',
    borderRadius: 5,
  },
});
