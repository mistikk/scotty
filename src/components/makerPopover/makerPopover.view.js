import React from 'react';
import { View, Text, Platform } from 'react-native';

// Component
import styles from './makerPopover.styles';
import { Button } from '../button';

const MakerPopoverView = ({
  id, name, distance, rating, handleOnPressBookmark,
}) => (
  <View style={styles.container}>
    <Text style={styles.title}>{name}</Text>
    <Text>{`Distance : ${parseInt(distance, 10)}`}</Text>
    <Text>{`Rating : ${rating}`}</Text>
    {(Platform.OS === 'android') ? (
      <Text>Click here to dd to bookmarks</Text>
    ) : (
      <Button
        style={styles.button}
        text="Add to bookmarks"
        onPress={() => handleOnPressBookmark(id)}
      />
    )}

  </View>
);

export default MakerPopoverView;
