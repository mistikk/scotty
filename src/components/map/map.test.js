import React from 'react';
import renderer from 'react-test-renderer';
import { Callout } from 'react-native-maps';
import { MarkerPopover } from '../makerPopover';

import Map from './map.view';

jest.mock('react-native-maps', () => {
  const React = require.requireActual('react');
  const MapView = require.requireActual('react-native-maps');

  class MockCallout extends React.Component {
    render() {
      return React.createElement('Callout', this.props, this.props.children);
    }
  }

  class MockMarker extends React.Component {
    render() {
      return React.createElement('Marker', this.props, this.props.children);
    }
  }

  class MockMapView extends React.Component {
    render() {
      return React.createElement('MapView', this.props, this.props.children);
    }
  }

  MockCallout.propTypes = MapView.Callout.propTypes;
  MockMarker.propTypes = MapView.Marker.propTypes;
  MockMapView.propTypes = MapView.propTypes;
  MockMapView.Marker = MockMarker;
  MockMapView.Callout = MockCallout;

  return MockMapView;
});

describe('Map Component', () => {
  const props = {
    latitude: 37.785834,
    loading: false,
    longitude: -122.406417,
    places: [
      {
        categories: [{ alias: 'cocktailbars' }, { alias: 'tacos' }],
        coordinates: {
          latitude: 37.7853008468227,
          longitude: -122.403918653727,
        },
        distance: 227.41363185164334,
        id: '_EncdQezAzcShATMFXL0dA',
        name: 'Tropisueno',
        rating: 4,
      },
    ],
  };

  it('> Callout onClick for Android platform', () => {
    const component = renderer.create(<Map {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('> button onClick function', () => {
    jest.mock('Platform', () => {
      const Platform = require.requireActual('Platform');
      Platform.OS = 'android';
      return Platform;
    });

    const mockCallback = jest.fn();
    const component = renderer.create(<Map {...props} addToBookmarks={mockCallback} />);

    component.root.findByType(Callout).props.onPress();
    expect(mockCallback.mock.calls[0][0]).toBe(props.places[0]);
  });

  it('> MarkerPopover onClick for iOS platform', () => {
    jest.mock('Platform', () => {
      const Platform = require.requireActual('Platform');
      Platform.OS = 'ios';
      return Platform;
    });

    const mockCallback1 = jest.fn();
    const component = renderer.create(<Map {...props} addToBookmarks={mockCallback1} />);

    component.root.findByType(MarkerPopover).props.handleOnPressBookmark(props.places[0].id);
    expect(mockCallback1.mock.calls[0][0]).toBe(props.places[0]);
  });
});
