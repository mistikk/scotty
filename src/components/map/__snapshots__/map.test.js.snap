// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map Component > Callout onClick for Android platform 1`] = `
<View>
  <MapView
    followsUserLocation={true}
    initialRegion={
      Object {
        "latitude": 37.785834,
        "latitudeDelta": 0.01,
        "longitude": -122.406417,
        "longitudeDelta": 0.01,
      }
    }
    showsMyLocationButton={true}
  >
    <Marker
      coordinate={
        Object {
          "latitude": 37.7853008468227,
          "longitude": -122.403918653727,
        }
      }
    >
      <Callout
        onPress={[Function]}
      >
        <View>
          <Text>
            Tropisueno
          </Text>
          <Text>
            Distance : 227
          </Text>
          <Text>
            Rating : 4
          </Text>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text>
              Add to bookmarks
            </Text>
          </View>
        </View>
      </Callout>
    </Marker>
  </MapView>
</View>
`;
